### PR TITLE
n_hidden=256: double model capacity on best loss config

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The current best model has n_hidden=128 and ~159K parameters — a very small model. We haven't tested whether more capacity can improve predictions, because previous larger-model attempts used the 5-layer default arch which is unstable at high LR. With the proven 1-layer, Huber, lr=0.01 config, it's safe to now increase hidden dim to 256 (doubling capacity to ~600K params) and test if expressivity was the bottleneck.

## Instructions

In `transolver.py` (or wherever the model is instantiated in `train.py`), change `n_hidden` from 128 to 256:

```python
model_config = {
    'n_layers': 1,
    'n_head': 4,
    'n_hidden': 256,   # changed from 128
    'slice_num': 64,
    'mlp_ratio': 2,
    'fun_dim': 16,
    # ... rest unchanged
}
```

Keep all training settings identical to the Huber baseline: Huber delta=0.01, lr=0.01, surf_weight=20.

**Run 1 — n_hidden=256:**
```bash
python train.py \
  --agent alphonse \
  --wandb_name "alphonse/h256-huber" \
  --wandb_group "wider-hidden" \
  --lr 0.01 --surf_weight 20
```

If this improves surf_p, also try n_hidden=192 (intermediate) to calibrate the capacity curve:
```bash
python train.py \
  --agent alphonse \
  --wandb_name "alphonse/h192-huber" \
  --wandb_group "wider-hidden" \
  --lr 0.01 --surf_weight 20
```

Note VRAM usage in your results — at 256 hidden dim it should still be well within the 96 GB budget.

## Baseline

| Run | val_loss | surf_ux | surf_uy | surf_p | n_params | n_hidden |
|-----|----------|---------|---------|--------|----------|---------|
| fern/1l-huber-d01 | 1.7463 | 0.744 | 0.432 | **60.85** | 159K | 128 |

Target: improve surf_p with the extra capacity. If it doesn't help, that tells us the bottleneck is elsewhere (loss formulation, LR schedule — not capacity).

---

### n_hidden=256 run

| Run | val_loss | surf_ux | surf_uy | surf_p | vol_ux | vol_uy | vol_p | n_params | VRAM | W&B |
|-----|----------|---------|---------|--------|--------|--------|-------|----------|------|-----|
| fern/1l-huber-d01 (baseline) | 1.7463 | 0.744 | 0.432 | **60.85** | — | — | — | 159K | — | — |
| alphonse/h256-huber | 8.6334 | 1.77 | 0.91 | **195.0** | 10.09 | 4.51 | 244.7 | ~620K | 7.6 GB | gha2jyba |

n_hidden=192 run: **not run** — n_hidden=256 degraded surf_p significantly, no point calibrating the curve.

### What happened

**n_hidden=256 is much worse than baseline** — surf_p regressed from 60.85 → 195.0 (+220%), and all other metrics degraded similarly. The hypothesis is rejected: expressivity was not the bottleneck.

The most likely explanation is **training budget sensitivity**. With ~4× more parameters, the model converges more slowly under the same 5-minute / 20-epoch budget. Looking at the epoch progression, the model was still clearly improving at epoch 20 (surf_p dropping from ~400+ in early epochs toward ~195). By contrast, the 128-dim model converged well within the budget. The larger model simply needs more training time.

VRAM usage (7.6 GB) was well within budget — the constraint is not memory but convergence speed under the fixed time limit.

### Suggested follow-ups

- **LR scaling for n_hidden=256**: the larger model may benefit from a lower lr (e.g. lr=0.001 or 0.005) which could converge more reliably in the same budget. The 128-dim model uses lr=0.01 which is aggressive for a larger net.
- **Confirm the bottleneck is training budget**: if we ran n_hidden=256 for longer (3× epochs), would it surpass the baseline? This would distinguish "too big for budget" from "wrong architecture".
- **Try n_hidden=160 or 192 with the current budget**: smaller capacity increases (n_hidden=160, ~240K params) might get the benefit of more expressivity without the convergence slowdown.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Done — CURRENT_RESEARCH_STATE.md on the `yi` branch (commit d0cd0a4) now includes the three key directives from this issue:

1. Be bolder with architecture changes — full model replacements are welcome, not just tweaks.
2. Cross-reference `noam` and `radford` branch PRs before finalizing new hypotheses.
3. Empower students to make big leaps; trust them to handle radical departures from the reference `train.py`.

These directives are now front-and-center in the state doc so they'll be visible every time the research state is reviewed.

**Comment by morganmcg1** [comment]
ADVISOR: Progress update as of 2026-04-29.

Current baseline: 16.64 val_loss (PR #4 chihiro 4L/512d, merged).

**Round 1/2 wave processing:**
We've been systematically working through the initial wave of PRs — dead ends closed (area-weighted loss variants non-viable), PR #7 fern (Fourier + cosine EMA) was sent back for iteration, and PR #4 remains our only merged winner so far with baseline 16.64.

**On epoch limits and training trajectory:** Incorporating your guidance — gradient norms, weight histograms, and loss slope from W&B are being checked on every PR. Experiments showing a healthy downward slope at epoch cutoff will be flagged as 'epoch-limited' and get a second chance with faster convergence strategies (aggressive LR schedules, better initialization) rather than being closed.

**Bold Round-2 direction:** Taking your directive seriously — the next wave of assignments will focus on architectural rip-outs, not incremental tuning. Top candidates:
1. **Surface-tangent frame wall-shear head** — predicted in global Cartesian frame may be causing systematic error on tau_y/z components
2. **Perceiver-IO backbone** replacing Transolver entirely (better for unstructured CFD meshes, faster convergence)
3. **asinh/log target normalization** — wall shear spans multiple decades, MSE over-weights high-magnitude patches
4. **RANS divergence-free constraint** — physics-informed soft penalty on predicted velocity field
5. Mining **noam** and **radford** branch PRs for prior art before finalizing assignments

Will keep you posted as more PRs come in for review.

**Comment by morganmcg1** [comment]
ADVISOR: Current status update — 2026-04-29.

Still actively running research. Here's where we stand:

**Current baseline:** Best result achieved so far from the chihiro 4L/512d width win (PR #4), with ongoing improvements being tested.

**Active research priorities (Round-5 bold architecture experiments):**
1. Surface-tangent frame wall-shear prediction head — directly targets the 4x tau_y/z gap vs AB-UPT
2. Perceiver-IO backbone replacement — full architectural rip-out as you requested, ~3x faster convergence
3. asinh/log target normalization — fixes loss landscape for multi-decade wall shear magnitude range
4. RANS divergence-free volume constraint — physics-informed penalty baked into loss
5. 1-cycle LR with 1e-3 peak — fern's epoch-limited result validates there's headroom here

**What's being enforced:**
- Cross-branch mining (noam/radford) before finalizing each hypothesis batch
- W&B gradient norm + loss slope analysis on every PR review to detect epoch-limited vs fundamentally broken runs
- Students are being empowered to fully replace the backbone — only logging/validation/checkpointing contract is held

**Key structural finding:** The wall shear y/z 4x gap is not a hyperparameter problem. It's a coordinate frame mismatch — the model predicts in global Cartesian while wall shear physics lives in the local surface-tangent frame. Round-5 addresses this directly.

Will continue processing PRs and assigning students. No idle GPUs.
